util/StringToTime.cc.lua

```diff
@@ -0,0 +1,123 @@
+{{/*
+  This command allows you to parse string to time. Is human friendly.
+
+  Recommended usage: As a part of a Larger Command. Can also be used standalone with -
+      Trigger: Command trigger with trigger `time`
+
+  Usage: 
+      String to be converted is fed to -> $timeString
+      Converted time is available in variable -> $timeConverted
+
+      Supported Syntax :   
+
+      Date:  Format 1 :  dd/mm/yyyy or dd.mm.yyyy or dd-mm-yyyy or dd,mm,yyyy
+             Format 2 :  String format with year mentioned with 4 digits and both short and long month names supported. Date components (i.e day , month , year) need not be present together. eg: 12 Feb 11:50 am , 2020 is supported.
+             Format 3 :  Today and tomorrow is supported.
+
+      TimeZone: By default timezone is UTC. If user has timezone set using "setz" command, timezone adjustment is also possible.
+
+      Time: Time is mentioned as hh:mm:ss or hh:mm or hh. May or may not be followed by am or pm.
+
+  Credits: Developed by Satty#9361
+*/}}
+
+{/* CONFIGURATION VALUES START */}}
+{{/*String to be converted is fed to $timeString variable*/}}
+{{$timeString := .StrippedMsg}}
+{{/* can be any string variable / string data but should contain only the time in the formats mentioned */}}
+{{/* CONFIGURATION VALUES END */}}
+
+{{/* Variable Declarations */}}
+{{$dTime := sdict "Day" currentTime.Day "Month" (toInt (printf "%d" currentTime.Month)) "Year" currentTime.Year  "Hour" currentTime.Hour "Min" currentTime.Minute "Sec" currentTime.Second}}
+{{$time := sdict "Day" $dTime.Day "Month" $dTime.Month "Year" $dTime.Year "Hour" 0 "Min" 0 "Sec" 0 }}
+{{$dateSet := false}} {{$timeSet := false}}
+{{$timeConverted := 0}}
+
+{{/* Actual Code */}}
+{{/* Fetching Dates */}}
+{{/* Fetching dates written in format dd.mm.yyyy or dd/mm/yyyy or dd-mm-yyyy or dd,mm,yyyy */}} 
+{{with reFindAllSubmatches `((\s|^)((\d{1,2})(\-|\.|\/|\,)(\d{1,2})(\-|\.|\/|\,)(\d{1,4}))(\s|$))` $timeString}}
+{{$time.Set "Day" (toInt (index (index . 0) 4))}}
+{{$time.Set "Month" (toInt (index (index . 0) 6))}}
+{{$time.Set "Year" (toInt (index (index . 0) 8))}}
+{{$dateSet = true}}
+{{else}}
+{{/* Fetching dates written as a string with both long or short month names supported. Date , Month and Year need not be present together but year must be written in full form(with 4 digits) eg: 20 sept 1am ,2019 is supported */}}
+     {{ with (reFindAllSubmatches `(?:[^a-z]|^)(jan((uary)?)|feb((ruary)?)|mar((ch)?)|apr((il)?)|may|jun(e?)|jul(y?)|aug((ust)?)|sep((t(ember)?)?)|oct((ober)?)|nov((ember)?)|dec((ember)?))(?:[^a-z]|$)` (lower $timeString) ) }}
+     {{$month := index (index . 0) 1}}
```

Comment:
The next if-else-if chain could be refactored into slicing month variable by 0 3 (first three chars) and indexing sdict of key shortened 3 char month name to numerical month. Sorry if this explanation wasn't too good, hopefully you understand :P If not, feel free to leave a comment.